sklearn/utils/__init__.py

```diff
@@ -81,6 +82,39 @@
 _IS_32BIT = 8 * struct.calcsize("P") == 32
 
 
+def _in_unstable_openblas_configuration():
+    """Return True if in an unstable configuration for OpenBLAS"""
+
+    # Import libraries which might load OpenBLAS.
+    import numpy  # noqa
+    import scipy  # noqa
+
+    modules_info = threadpool_info()
+
+    open_blas_used = any(info["internal_api"] == "openblas" for info in modules_info)
+    if not open_blas_used:
+        return False
+
+    # OpenBLAS < 0.3.13 causes segfault for Prescott architecture, see:
+    # https://github.com/scikit-learn/scikit-learn/issues/21361 # noqa
+    openblas_stable_version = LooseVersion("0.3.13")
+    for info in modules_info:
+        if info["internal_api"] != "openblas":
+            continue
+        openblas_version = info.get("version")
+        openblas_architecture = info.get("architecture")
+        if openblas_version is None or openblas_architecture is None:
+            # Cannot be sure that we use a good version of OpenBLAS
+            # Hence assume bad configuration:
+            return True
```

Comment:
could we start without this part and see if it's enough?

Reply:
In `threadpoolctl< 3.0.0`, architectures were not reported, hence this prudent case. Still, I will try to remove it.

Reply:
That's a fair point. But the question is if we use older threadpoolctl in our CI. Since this is used only in our test suit and CI, I don't think we have to support older versions.

Reply:
Probably, but removing it might break some users' setup locally which uses threadpoolctl < 3.0.0 when they run tests for unrelated reasons they might take time to understand.

Reply:
This situation might be rare but might happen. What do you think?